cli/dcoscli/cluster/main.py

```diff
@@ -106,8 +133,11 @@ def _list(json_, attached):
     :rtype: None
     """
 
-    clusters = [c.dict() for c in cluster.get_clusters()
+    # Query for linked clusters only when listing all clusters.
+    linked = not attached
+    clusters = [c.dict() for c in cluster.get_clusters(linked)
                 if not attached or c.is_attached()]
```

Comment:
The logic above here is confusing. Can you change it to make it more clear about what it si doing. I'm still not sure after reading it a few times. Otherwise, can you make the comment more clear.

Reply:
I also found it unclear at the beginning, that's why I made a variable before and added a comment.

This is the `list` command, the behavior here being that `dcos cluster list --attached` should not query for linked cluster but only return the current one. That's why I added a boolean argument to `get_clusters()`. I'll clarify the comment

Reply:
When I read the line:
```
cluster.get_clusters(linked)
```
that implies to me that I am getting the set of clusters that are linked (and no others)

Or if we changed it to:
```
cluster.get_clusters(not attached)
```
that implies to me that we are getting the list of non-attached clusters.

which then seems weird to combine with the condition for `if not attached or c.is_attached()` below it.

The whole thing is just confusing to read from left to right and should be made clearer for someone without having to look at the semantics of what the argument to `cluster.get_clusters(...)` is.

Reply:
Here I could make the variable more explicit (eg. `with_links`, `attached_only`) and clarify the comment. Would that help?

Otherwise I'll switch to a standard loop instead of a list comprehension.

Reply:
Ignoring the implementation for the minute, let me make sure I understand what this is trying to do. In (really) long form, would it be something like:
```
if attached:
    clusters = [c.dict() for c in cluster.get_clusters() if c.is_attached()]
else:
    clusters = clusters = [c.dict() for c in cluster.get_clusters(True)]
```
If so, this feels much clearer to me than trying to wrap that all into a single statement (and is actually faster since there are less comparisons per iteration in the comprehension loop).